### ISSUE TITLE
GPUTreeShap shap-loss values different from TreeShap

### ISSUE THREAD

**Issue by gtcf** [shap/shap#1726]
It seems that GPUTreeShap gives different shap values from TreeShap, when using 
feature_perturbation="interventional", 
model_output="log_loss".
```
from sklearn.model_selection import train_test_split
import lightgbm as lgb
import numpy as np
from pprint import pprint
import shap
import time

# # print the JS visualization code to the notebook
# shap.initjs()

## Load dataset

X,y = shap.datasets.adult()
X_display,y_display = shap.datasets.adult(display=True)

# create a train/test split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=7)
d_train = lgb.Dataset(X_train, label=y_train)
d_test = lgb.Dataset(X_test, label=y_test)

## Train the model

params = {
    "max_bin": 512,
    "learning_rate": 0.05,
    "boosting_type": "gbdt",
    "objective": "binary",
    "metric": "binary_logloss",
    "num_leaves": 10,
    "verbose": -1,
    "min_data": 100,
    "boost_from_average": True
}

model = lgb.train(params, d_train, 10000, valid_sets=[d_test], early_stopping_rounds=50, verbose_eval=1000)

## Explain predictions

# Here we use the Tree SHAP implementation integrated into Light GBM to explain the entire dataset (32561 samples).
train_sample = X_train.sample(1000)
loss_explainer = shap.TreeExplainer(model, train_sample, 
                            feature_perturbation="interventional", 
                                  model_output="log_loss")
start = time.process_time()
shap_loss_values = loss_explainer.shap_values(X_test, y_test)

print("\nSHAP values in CPU were estimated in %d sec." % (time.process_time() - start))

gpu_loss_explainer = shap.GPUTreeExplainer(model, train_sample, 
                            feature_perturbation="interventional", 
                                  model_output="log_loss")
start = time.process_time()
shap_loss_values_gpu = gpu_loss_explainer.shap_values(X_test, y_test)
print("\nSHAP values in GPU were estimated in %d sec." % (time.process_time() - start))

if not np.isclose(shap_loss_values_gpu, shap_loss_values).all():
    print("\nCAUTION: different SHAP values were obtained using GPU:")
    print(np.around(shap_loss_values_gpu - shap_loss_values, 2))

```

> 
> Training until validation scores don't improve for 50 rounds
> Early stopping, best iteration is:
> [644]	valid_0's binary_logloss: 0.278029
> 
>  99%|===================| 6434/6513 [00:54<00:00]        
> 
> 
> SHAP values in CPU were estimated in 53 sec.
> 
> SHAP values in GPU were estimated in 1 sec.
> 
> CAUTION: different SHAP values were obtained using GPU:
> [[ 0.21  0.03  0.   ... -0.19  0.02  0.  ]
>  [ 1.37  0.62  1.24 ... -0.1   1.48  0.02]
>  [-1.45 -0.01  0.01 ... -0.06 -0.1   0.  ]
>  ...
>  [-2.55  0.11  0.11 ... -0.08 -0.79 -0.01]
>  [-0.76 -0.11 -0.13 ... -0.06  0.06  0.26]
>  [ 0.57  0.01  0.13 ... -0.02  0.11  0.02]]
>
